api/dataset.go

```diff
@@ -353,25 +354,40 @@ func (api *DatasetAPI) deleteDataset(w http.ResponseWriter, r *http.Request) {
 		currentDataset, err := api.dataStore.Backend.GetDataset(datasetID)
 
 		if err == errs.ErrDatasetNotFound {
-			log.InfoCtx(ctx, "cannot delete dataset, it does not exist", logData)
+			log.InfoCtx(ctx, "delete dataset endpoint: cannot delete dataset, it does not exist", logData)
 			return errs.ErrDeleteDatasetNotFound
 		}
 
 		if err != nil {
-			log.ErrorCtx(ctx, errors.WithMessage(err, "failed to run query for existing dataset"), logData)
+			log.ErrorCtx(ctx, errors.WithMessage(err, "delete dataset endpoint: failed to run query for existing dataset"), logData)
 			return err
 		}
 
 		if currentDataset.Current != nil && currentDataset.Current.State == models.PublishedState {
-			log.ErrorCtx(ctx, errors.WithMessage(errs.ErrDeletePublishedDatasetForbidden, "unable to delete a published dataset"), logData)
+			log.ErrorCtx(ctx, errors.WithMessage(errs.ErrDeletePublishedDatasetForbidden, "delete dataset endpoint: unable to delete a published dataset"), logData)
 			return errs.ErrDeletePublishedDatasetForbidden
 		}
 
+		// Find any editions associated with this dataset
+		editionDocs, err := api.dataStore.Backend.GetEditions(currentDataset.ID, "")
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(errs.ErrEditionsNotFound, "delete dataset endpoint: unable to find the dataset editions"), logData)
+			return errs.ErrEditionsNotFound
+		}
+
+		// Then delete them
+		for i := range editionDocs.Items {
```

Comment:
Is this valid `Go`? I thought the range returned 2 args `index` && `item`? I could be wrong.

Reply:
it's valid - might be a bit of of a python-istic way of writing it, not sure :)

makes more sense here: https://play.golang.org/p/KEx4z-3hxOx

happy to change it if you like.

Reply:
it's valid - might be a bit of of a python-istic way of looking at it though, not sure.

makes more sense here: https://play.golang.org/p/KEx4z-3hxOx

happy to change it if you think it'd be clearer.

Reply:
https://gobyexample.com/range - yes, to just take the keys or the index it's valid/advertised as a thing :) whether we choose to do it is certainly another matter, but I think its dotted around a few places in our code already

Reply:
its very similar to the optional return when you read from a map, e.g.:

`value := mappything[key]` or `value, ok := mappything[key]` which are both valid, but the boolean return is optional (for checking if that key exists mostly)